includes/database/class-edd-db-table-discounts.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Discounts Table: EDD_DB_Table_Discounts class
+ *
+ * @package Plugins/EDD/Database/Object
+ */
+
+// Exit if accessed directly
+defined( 'ABSPATH' ) || exit;
+
+if ( class_exists( 'WP_DB_Table' ) ) :
+/**
+ * Setup the global "edd_discounts" database table
+ *
+ * @since 3.0.0
+ */
+final class EDD_DB_Table_Discounts extends WP_DB_Table {
+
+	/**
+	 * @var string Table name
+	 */
+	protected $name = 'edd_discounts';
+
+	/**
+	 * @var string Database version
+	 */
+	protected $version = 201801060001;
+
+	/**
+	 * @var boolean This is not a global table
+	 */
+	protected $global = false;
+
+	/**
+	 * Setup the database schema
+	 *
+	 * @since 3.0.0
+	 */
+	protected function set_schema() {
+		$this->schema = "id bigint(20) unsigned NOT NULL auto_increment,
+			name varchar(200) NOT NULL default '',
+			code varchar(50) NOT NULL default '',
+			status varchar(20) NOT NULL default '',
+			type varchar(20) NOT NULL default '',
+			scope varchar(20) NOT NULL default 'all',
+			amount mediumtext NOT NULL default '',
+			description longtext NOT NULL default '',
+			max_uses bigint(20) unsigned NOT NULL default 0,
+			use_count bigint(20) unsigned NOT NULL default 0,
```

Comment:
Do we need `bigint` for these use counters? That's a huge number and if I understand correctly requires more storage space.

Reply:
Probably not, but the compelling reason to stick with `bigint(20)` everywhere is because it's what WordPress uses almost everywhere else for numeric values.

WordPress also uses `int(11)` in some places that are `rating` or `karma` related, but they aren't actually used in core anymore. `user_status` still uses this, incorrectly (should be a string key with an API like posts.)

See this post for more on storage: https://stackoverflow.com/questions/4769416/mysql-bigint-vs-int

> The difference is purely in the maximum value which can be stored

4-byte vs. 8-byte per character. Since everything is UTF8 going forward, it's probably best to stay with `bigint` but the `(20)` for `ZEROFILL` is a stylistic decision that I don't feel strongly about.